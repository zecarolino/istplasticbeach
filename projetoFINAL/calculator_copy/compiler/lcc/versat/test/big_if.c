void putchar(int ch) { asm("\trdw R11\n\taddi 2\n\twrw RB\n\trdwb\n\trdwb\n\twrw CPRT_BASE\n"); }
int x = 3, y = 3, z = 3;
int main() {
	if (x < 0) return 0;
	if (y < 0) return 0;
	if (z < 0) return 0;
	if (x + y <= z) {
		putchar('0');
		putchar('\n');
		return 1;
	}
	if (y + z <= x) {
		putchar('0');
		putchar('\n');
		return 1;
	}
	if (z + x <= y) {
		putchar('0');
		putchar('\n');
		return 1;
	}
	if (x == y) {
		if (y == z) putchar('3');
		else putchar('2');
		putchar('\n');
		return 1;
	}
	if (z == y) {
		putchar('2');
		putchar('\n');
		return 1;
	}
	if (z == x) {
		putchar('2');
		putchar('\n');
		return 1;
	}
	putchar('1');
	putchar('\n');
	return 1;
}
