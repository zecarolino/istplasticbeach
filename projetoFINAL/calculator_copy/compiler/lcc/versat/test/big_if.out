VCD info: dumpfile xtop.vcd opened for output.
3
1
Catched Trap at time  708 clock cycles:
